python/src/aiconfig/editor/server/server.py

```diff
@@ -122,42 +124,48 @@ async def run() -> FlaskPostResponse:
     try:
         result = await state.aiconfig.run(prompt_name, options=inference_options)  # type: ignore
         LOGGER.debug(f"Result: {result=}")
-        return HttpPostResponse(
+        return HttpResponseWithAIConfig(
             message="Done",
             aiconfig=state.aiconfig,
         ).to_flask_format()
-        # return {"message": "Done", "output": result_text}, 200
     except Exception as e:
         err: Err[str] = core_utils.ErrWithTraceback(e)
         LOGGER.error(f"Failed to run: {err}")
-        return HttpPostResponse(
+        return HttpResponseWithAIConfig(
             message=f"Failed to run: {err}",
             code=400,
             aiconfig=None,
         ).to_flask_format()
 
 
 @app.route("/api/add_prompt", methods=["POST"])
-def add_prompt() -> FlaskPostResponse:
+def add_prompt() -> FlaskResponse:
     state = get_server_state(app)
     request_json = request.get_json()
     try:
         LOGGER.info(f"Adding prompt: {request_json}")
         state.aiconfig.add_prompt(**request_json)  # type: ignore
-        return HttpPostResponse(
+        return HttpResponseWithAIConfig(
             message="Done",
             aiconfig=state.aiconfig,
         ).to_flask_format()
     except Exception as e:
         err: Err[str] = core_utils.ErrWithTraceback(e)
         LOGGER.error(f"Failed to add prompt: {err}")
-        return HttpPostResponse(
+        return HttpResponseWithAIConfig(
             message=f"Failed to add prompt: {err}",
             code=400,
             aiconfig=None,
         ).to_flask_format()
 
 
+@app.route("/api/list_models", methods=["GET"])
+def list_models() -> FlaskResponse:
+    out: list[str] = ModelParserRegistry.parser_ids()  # type: ignore
```

Comment:
Is this the same as `ModelParserRegistry.display_parsers().keys()`? parser ids vs model names is still a bit confusing to me.

Also, can we support searching by substring here? 

I had done:
```
request_json = request.get_json()
all_models: list[core_utils.JSONValue] = list(ModelParserRegistry.display_parsers().keys())
search = request_json.get("search", "")
models = [m for m in all_models if search in m]
```
which is currently erroring on bad request format but the general concept should be there (and I can scrap it in favour of your much better python code)

Reply:
> parser ids vs model names is still a bit confusing to me.

Me too, dont know off the top of my head. @Ankush-lastmile ?

Reply:
Ok just checked, this returns the mapping keys, which are the model names.